R/calc_dist_params.R

```diff
@@ -47,26 +47,50 @@
 #'   sample_size = 25
 #' )
 #' }
-calc_dist_params <- function(prob_dist,
+calc_dist_params <- function(prob_dist, # nolint cyclocomp
+                             prob_dist_params,
                              summary_stats,
-                             sample_size = NA) {
-  # extract mean and sd to see if conversion is possible
-  mean_sd <- c(
-    mean = summary_stats$centre_spread$mean,
-    sd = summary_stats$centre_spread$sd
-  )
+                             sample_size) {
+  if (is.na(prob_dist)) {
+    message(
+      "No adequate summary statistics available to calculate the parameters ",
+      "of the ", prob_dist, " distribution"
+    )
+    return(NA)
+  }
 
-  # convert percentile names to numbers
-  percentiles <- get_percentiles(summary_stats$quantiles)
+  is_percentiles <- checkmate::test_numeric(
+    summary_stats$quantiles,
+    all.missing = FALSE,
+    names = "unique",
+    null.ok = FALSE
+  )
+  if (is_percentiles) {
+    # convert percentile names to numbers
+    percentiles <- get_percentiles(summary_stats$quantiles)
+  } else {
+    percentiles <- NA
+  }
 
   # extract median and range to calculate parameters as third choice
   median_range <- c(
-    median = summary_stats$centre_spread$median,
+    median = summary_stats$median,
     unlist(summary_stats$range)
   )
 
-  if (!anyNA(mean_sd)) {
-    # unlist and remove NAs
+  # extract dispersion
+  disp <- unname(prob_dist_params[names(prob_dist_params) == "dispersion"])
+  median_disp <- c(median = summary_stats$median, dispersion = disp)
+
+  # extract mean and sd
+  mean_sd <- c(summary_stats$mean, summary_stats$sd)
+
+  is_mean_sd <- checkmate::test_numeric(mean_sd, any.missing = FALSE, len = 2)
+  is_median_disp <- checkmate::test_numeric(median_disp, len = 2)
+  is_median_range <- checkmate::test_numeric(median_range, len = 3) &&
```

Comment:
Sorry to nitpick - you might want to ensure these are within an acceptable range, e.g. positive values and finite. If this function is internal, prefer checking at the top level.

Reply:
Thanks, resolved in b2fea45a19566db8b75d0a9ef7eb603b0dd5a75e.